test/test_pty.rs

```diff
@@ -0,0 +1,36 @@
+use nix::pty::openpty;
+use nix::unistd::{read, write, close};
+
+#[test]
+fn test_openpty() {
+    // TODO: figure out the right termios settings to pass in here
```

Comment:
It doesn't really matter what flags you set so long as the function gets tested with `None` and `Some(_)` types really. If the CR thing works, go ahead and add a test that tests that.